src/main/java/org/jabref/logic/importer/MergeEntriesHelper.java

```diff
@@ -0,0 +1,87 @@
+
+package org.jabref.logic.importer;
+
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableChangeType;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.FieldFactory;
+import org.jabref.model.entry.types.EntryType;
+
+public class MergeEntriesHelper {
```

Comment:
Isn't `@AllowedToUseSwing` missing as I wrote before in the reivew comments?

Reply:
I need to admit that when I saw your earlier comment about moving "this whole method can go into logic", I didn't truly understand the full implications. I thought simply moving the method to the logic layer was sufficient, and missed the importance of the "@AllowedToUseSwing". I'll remove the Swing dependency and update the code later. Sorry for my oversight.

Reply:
> I'll remove the Swing dependency

Please think a bit.  I think, i posted the following example:

    @AllowedToUseSwing("UndoableUnabbreviator and UndoableAbbreviator requires Swing Compound Edit in order test the abbreviation and unabbreviation of journal titles")

You need undo/redo. This is OK then.

Only not OK is the background task.